app/src/main/java/com/quark/client/pages/Chat.kt

```diff
@@ -34,8 +34,11 @@ fun Chat(props: ChatProps) {
         mutableStateOf<List<QuarkMessage>>(emptyList())
     }
 
-    LaunchedEffect(key1 = props.fromId) {
-        messages = props.messages.getMessagesToFrom(props.toId, props.fromId)
+    LaunchedEffect(key1 = props.conversationID, key2 = props.users) {
+        props.messages.getConversation(props.conversationID).map { message ->
+            val usernameFrom = props.users.getUserProfileById(message.userFrom)?.username.toString()
```

Comment:
we need to move this out of map somehow, so we get usernames of user1 and user2 once when you open the convo, and not for every message.

this'll cause issues because we're limited on reads in firebase